pkg/3scale/amp/operator/apicast_options_provider.go

```diff
@@ -532,3 +575,130 @@ func (a *ApicastOptionsProvider) setPodTemplateAnnotations() {
 	a.apicastOptions.StagingPodTemplateAnnotations = a.apimanager.Spec.Apicast.StagingSpec.Annotations
 	a.apicastOptions.ProductionPodTemplateAnnotations = a.apimanager.Spec.Apicast.ProductionSpec.Annotations
 }
+
+func (a *ApicastOptionsProvider) getOpenTelemetryStagingConfig(ctx context.Context) (component.OpentelemetryConfig, error) {
+	res := component.OpentelemetryConfig{
+		Enabled: false,
+	}
+
+	if a.apimanager.Spec.Apicast.StagingSpec.OpenTelemetry != nil && a.apimanager.Spec.Apicast.StagingSpec.OpenTelemetry.Enabled != nil {
+		res = component.OpentelemetryConfig{
+			Enabled: *a.apimanager.Spec.Apicast.StagingSpec.OpenTelemetry.Enabled,
+		}
+	} else {
+		return res, nil
+	}
+
+	if !res.Enabled {
+		return res, nil
+	}
+
+	// In the APIcast CR validation step it is checked that when enabled, the secret ref is not nil
+	// Adding this to avoid panics
+	if a.apimanager.Spec.Apicast.StagingSpec.OpenTelemetry.TracingConfigSecretRef == nil || a.apimanager.Spec.Apicast.StagingSpec.OpenTelemetry.TracingConfigSecretRef.Name == "" {
+		fldPath := field.NewPath("spec").Child("apicast").Child("stagingSpec").Child("openTelemetry").Child("tracingConfigSecretRef")
+		errors := append(field.ErrorList{}, field.Invalid(fldPath, a.apimanager.Spec.Apicast.StagingSpec.OpenTelemetry.TracingConfigSecretRef, "tracing config secret name is empty"))
+		return res, errors.ToAggregate()
```

Comment:
```suggestion
		errs := append(field.ErrorList{}, field.Invalid(fldPath, a.apimanager.Spec.Apicast.StagingSpec.OpenTelemetry.TracingConfigSecretRef, "tracing config secret name is empty"))
		return res, errs.ToAggregate()
```

There a few instances in this file where the variable name collides with the `errors` imported package name. It's probably a good idea to fix now if possible.